internal/app/machined/pkg/system/services/registry/registry.go

```diff
@@ -0,0 +1,248 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package registry provides a simple container registry service.
+package registry
+
+import (
+	"bytes"
+	"cmp"
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/containerd/containerd/v2/core/content"
+	"github.com/distribution/reference"
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/siderolabs/gen/xerrors"
+	"go.uber.org/zap"
+)
+
+// NewService creates a new instance of the registry service.
+func NewService(root string, logger *zap.Logger) *Service {
```

Comment:
let's add multiple roots (probably best if we can inject them on the fly? or the service itself can read that from the resource, but that can come later)

Reply:
this probably needs to be added

Reply:
it was added, but via MultiFS iterator (see the example in the service start function)

Reply:
oh i thought I commented on the healthcheck part, sorry